sys/zuse.hoon

```diff
@@ -8135,6 +8161,54 @@
         (~(put by tuf) q.i.dom p.i.dom)
       $(dom t.dom)
     --
+  ::  |snap:dawn restore from snapshot
+  ::
+  ++  snap
+    !:
+    |%
+    ::  +bloq:snap:dawn: extract block number
+    ::
+    ++  bloq
+      |=  snap=snapshot:jael
+      ^-  @ud
+      latest-block.etn.snap
+    ::  +czar:snap:dawn: extract galaxy table
+    ::
+    ++  czar
+      |=  snap=snapshot:jael
+      ^-  (map ship [=life =pass])
+      %-  malt
+      %+  turn  (gulf 0 255)
+      |=  gal=@
+      ^-  [ship [life pass]]
+      :-  gal
+      ~|  czar-gal=gal
+      [life pass]:(need net:(~(got by hul.eth.snap) gal))
+    ::  +hull:snap:dawn: extract ship's contract state
+    ::
+    ++  hull
+      |=  [who=ship snap=snapshot:jael]
+      ^-  hull:constitution:ethe
+      =/  res  (~(get by hul.eth.snap) who)
+      ?~  res
+        ~&  ['hull not found in snapshot; can\'t verify' who=who]
+        !!
+      u.res
+    ::  +turf:snap:dawn: extract network domains
+    ::
+    ++  turf
+      |=  snap=snapshot:jael
+      ^-  (list ^turf)
+      %+  murn
+        ^-  (list (pair))
+        %+  murn
+          ^-  (list @t)
+          ~[pri sec ter]:dns.eth.snap
+        |=  dom=@t
+        ^-  (unit (pair))
+        (rush dom thos:de-purl:html)
+      |=([* a=*] ((soft ^turf) a))
```

Comment:
For compatibility with `+turf:take:dawn`, this should filter out duplicates. Also not pressing.